File: solutions/001.two-sum/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
const twoSum = (nums, target) => {
  let set = new Set()

  let len = nums.length
  for (let i = 0; i < len; i++) {
    let item = nums[i]

    if (set.has(item)) return [i, nums.indexOf(target - item)].sort()
    else set.add(target - item)
  }
}

File: solutions/001.two-sum/two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        s = set()
        ret = [None, None]
        for i, n in enumerate(nums):
            if n in s:
                ret[1] = i
                ret[0] = nums.index(target - n)
                break
            s.add(target - n)
            
        return ret